pojo/src/main/java/com/spotify/hamcrest/pojo/MethodReference.java

```diff
@@ -21,10 +21,19 @@
 package com.spotify.hamcrest.pojo;
 
 import java.io.Serializable;
-import java.util.function.Function;
 
 /**
- * Adds serialization marker to {@code Function}.
+ * An interface for serializable method references. It is only valid to construct instances of this
```

Comment:
"only valid" as in the code won't compile, or the library will break in unexpected ways?